docs/reference/components/sinks/syslog.cue

```diff
@@ -0,0 +1,200 @@
+package metadata
+
+components: sinks: syslog: {
+	title: "Syslog"
+
+	classes: {
+		commonly_used: true
+		delivery:      "best_effort"
+		development:   "beta"
+		egress_method: "stream"
+		service_providers: []
+		stateful: false
+	}
+
+	features: {
+		buffer: enabled:      true
+		healthcheck: enabled: true
+		send: {
+			compression: enabled: false
+			encoding: enabled: false
+			send_buffer_bytes: {
+				enabled:       true
+				relevant_when: "mode = `tcp` or mode = `udp`"
+			}
+			keepalive: enabled: true
+			request: enabled:   false
+			tls: {
+				enabled:                true
+				can_enable:             true
+				can_verify_certificate: true
+				can_verify_hostname:    true
+				enabled_default:        false
+			}
+			to: {
+				service: services.syslog
+
+				interface: {
+					socket: {
+						api: {
+							title: "Syslog"
+							url:   urls.syslog
+						}
+						direction: "outgoing"
+						protocols: ["tcp", "udp", "unix"]
+						ssl: "required"
+					}
+				}
+			}
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration:  sinks.socket.configuration & {
+	    "type": "type": string: enum: syslog: "The type of this component."
+		rfc5424: {
+			common: true
+			description: "If this is set to `true` message will be formatted according to the RFC5424 else the RFC3164 will be used."
```

Comment:
Do you think it is better to default to RFC3164? I think RFC5424 would be preferred.

Reply:
The default value for `rfc5424` is `true`, but now I realize that having a `rfc3164` setting with a default value set to `false` is more explicit for people wanting to use the older formating. I updated the option accordingly.

Reply:
What would you think of making this an actual enum in the config?

For example: https://github.com/timberio/vector/blob/a21687c82a619ecafb24a80fcb939ae9015e7952/docs/reference/components/sources/aws_s3.cue#L62-L76

We could default to `rfc5424`. I think this would leave the door open to supporting more encodings in the future rather than a boolean flag.

Reply:
Indeed that make sense, if the rfc5848 (same as rfc5424 + extra structured data to sign the message) has to be supported it will be a convenient way of keeping the same kind of config.